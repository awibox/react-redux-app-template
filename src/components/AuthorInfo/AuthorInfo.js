import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ImmutablePropTypes from 'react-immutable-proptypes';
// Components
import { Button, Card } from 'tigerspack';
// Styles
import styles from './AuthorInfo.scss';

export default class AuthorInfo extends Component {
  static propTypes = {
    author: ImmutablePropTypes.contains({
      avatar_url: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      location: PropTypes.string.isRequired,
      company: PropTypes.string.isRequired,
      bio: PropTypes.string.isRequired,
      html_url: PropTypes.string.isRequired,
    }),
  };

  buttonClick = (url) => window.open(url);

  render() {
    const { author } = this.props;
    return (
      <Card theme={'light'} className={styles.authorInfo}>
        <div className={styles.logo} style={{ backgroundImage: `url(${author.get('avatar_url')})` }} />
        <div className={styles.title}>{author.get('name')}</div>
        <div className={styles.titleVersion}>{author.get('location')}</div>
        <div className={styles.infoText}>{author.get('bio')}</div>
        <div className={styles.button}>
          <Button icon={<i className="fa fa-github"/>} onClick={() => this.buttonClick(author.get('html_url'))}>
            Github page
          </Button>
        </div>
      </Card>
    );
  }
}
