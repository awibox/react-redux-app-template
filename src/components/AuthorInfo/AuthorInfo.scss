.authorInfo {
  text-align: center;
  display: flex;
  flex-flow: column;
  justify-content: center;
}

.logo {
  width: 150px;
  height: 150px;
  background-size: cover;
  background-repeat: no-repeat;
  margin: 30px auto;
}

.title {
  line-height: 56px;
  font-size: 56px;
  font-weight: 100;
  vertical-align: top;
  padding-bottom: 10px;
}

.titleVersion {
  font-size: 18px;
  line-height: 22px;
  margin-left: 5px;
}

.infoText {
  padding: 20px;
  font-size: 20px;
  color: #333;
  font-weight: 100;
  line-height: 30px;
  margin-bottom: 10px;
}

.button {
  padding-bottom: 20px;
}
