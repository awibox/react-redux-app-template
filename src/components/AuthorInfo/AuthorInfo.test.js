import React from 'react';
import { Map } from 'immutable';
import { shallow } from 'enzyme';
import AuthorInfo from './AuthorInfo';

const mockData = Map({
  avatar_url: 'http://localhost:8888/avatar',
  name: 'John',
  location: 'USA',
  company: 'Google',
  bio: 'Addition information',
  html_url: 'http://localhost:8888/url',
});

describe('AuthorInfo', () => {
  it('should render correctly', () => {
    const component = shallow(<AuthorInfo author={mockData} />);
    expect(component).toMatchSnapshot();
  });
});
