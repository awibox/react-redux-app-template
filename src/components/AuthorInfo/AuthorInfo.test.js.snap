// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorInfo should render correctly 1`] = `
<Card
  className="authorInfo"
  padding={15}
  theme="light"
>
  <div
    className="logo"
    style={
      Object {
        "backgroundImage": "url(http://localhost:8888/avatar)",
      }
    }
  />
  <div
    className="title"
  >
    John
  </div>
  <div
    className="titleVersion"
  >
    USA
  </div>
  <div
    className="infoText"
  >
    Addition information
  </div>
  <div
    className="button"
  >
    <Button
      icon={
        <i
          className="fa fa-github"
        />
      }
      onClick={[Function]}
    >
      Github page
    </Button>
  </div>
</Card>
`;
